pkg/guacanalytics/patchPlanning_test.go

```diff
@@ -0,0 +1,481 @@
+			startType:      "conan",
+			startNamespace: "openssl.org",
+			startName:      "openssl",
+			startVersion:   ptrfrom.String("3.0.3"),
+			maxDepth:       10,
+			expectedLen:    6,
+			expectedPkgs:   []string{"top", "deb", "conan"},
+			graphInput:     "isDependencySimpleGraph",
+		},
+		{
+			name:           "2:  one level of dependencies, no stopID and no limiting maxDepth",
+			startType:      "deb",
+			startNamespace: "ubuntu",
+			startName:      "dpkg",
+			startVersion:   ptrfrom.String("1.19.0"),
+			maxDepth:       10,
+			expectedLen:    4,
+			expectedPkgs:   []string{"top", "deb"},
+			graphInput:     "isDependencySimpleGraph",
+		},
+		{
+			name:           "3: two levels of dependencies, a stopID at the first level and no limiting maxDepth",
+			startType:      "conan",
+			startNamespace: "openssl.org",
+			startName:      "openssl",
+			startVersion:   ptrfrom.String("3.0.3"),
+			stopType:       ptrfrom.String("deb"),
+			stopNamespace:  "ubuntu",
+			stopName:       "dpkg",
+			stopVersion:    ptrfrom.String("1.19.0"),
+			maxDepth:       10,
+			expectedLen:    4,
+			expectedPkgs:   []string{"deb", "conan"},
+			graphInput:     "isDependencySimpleGraph",
+		},
+		{
+			name:           "4: two levels of dependencies, no stopID and a limiting maxDepth at the first level",
+			startType:      "conan",
+			startNamespace: "openssl.org",
+			startName:      "openssl",
+			startVersion:   ptrfrom.String("3.0.3"),
+			maxDepth:       1,
+			expectedLen:    4,
+			expectedPkgs:   []string{"deb", "conan"},
+			graphInput:     "isDependencySimpleGraph",
+		},
+		{
+			name:           "5: isDependency indirect dependency",
+			startType:      "bottom",
+			startNamespace: "bottomns",
+			startName:      "bottompkg",
+			startVersion:   ptrfrom.String("1.19.0"),
+			maxDepth:       10,
+			expectedLen:    8,
+			expectedPkgs:   []string{"top", "deb", "conan", "bottom"},
+			graphInput:     "isDependencySimpleGraph",
+		},
+		{
+			name:           "6: isDependency no dependents returns no extra",
+			startType:      "top",
+			startNamespace: "topns",
+			startName:      "toppkg",
+			startVersion:   ptrfrom.String("1.19.0"),
+			maxDepth:       10,
+			expectedLen:    2,
+			expectedPkgs:   []string{"top"},
+			graphInput:     "isDependencySimpleGraph",
+		},
+		{
+			name:           "7: direct isDependency not included in range",
+			startType:      "extraType",
+			startNamespace: "extraNamespace",
+			startName:      "extraName",
+			startVersion:   ptrfrom.String("1.19.0"),
+			maxDepth:       10,
+			expectedLen:    2,
+			expectedPkgs:   []string{"extraType"},
+			graphInput:     "isDependencyNotInRangeGraph",
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(fmt.Sprintf("Test case %s\n", tt.name), func(t *testing.T) {
+			err = ingestTestData(ctx, gqlclient, tt.graphInput)
+
+			if err != nil {
+				t.Errorf("Error ingesting test data: %s", err)
+				return
+			}
+
+			var startIDs []*string
```

Comment:
this should probably just be a string and then if the return from `getPackageIDs` is more than 1 then it should error out and not continue, then you can make the assumption going forward based on the type and not having to deal with addressing the 0th entry each time, which makes it so you have an implicit contract on the slice throughout the code. 

E.g. instead of *[]string where we need to assert it is non-nil and len==1, if we see *string, the type tells us that it is a singleton